### PR TITLE
Drop support for Python 3.6

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

Now that Qiskit Aer 0.10.0 has been released that Qiskit no longer
supports python 3.6 (in the lead up to adding support for python 3.10).
This commit officially drops the support for python 3.6 and makes python
3.7 required to install qiskit-aer. It also removes all the ci jobs
still using 3.6 both for testing and building wheels.

### CODE REVIEW

**Comment by mtreinish** [comment]
> Perhaps updates to the available tox environments as well?

Done in the most recent revision.

**Review by jakelishman — DISMISSED** [review]
These are all the references to Python 3.6 I could find too - looks good to me.

**Comment by jakelishman** [comment]
Oh wait, we probably need an update release note still.

**Comment by mtreinish** [comment]
> Oh wait, we probably need an update release note still.

Added in: https://github.com/Qiskit/qiskit-aer/pull/1430/commits/f1c65be25e98f7198a123dd0688854a2d5f29add

File: releasenotes/notes/drop-python36-61553302523fa240.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    Running qiskit-aer with Python 3.6 is no longer supported. Python >= 3.7
+    is no required to install and run qiskit-aer.
```
Comment (jakelishman) [inline releasenotes/notes/drop-python36-61553302523fa240.yaml:5]:
```suggestion
    is now required to install and run qiskit-aer.
```

**Review by jakelishman — APPROVED** [review]

**Review by jakelishman — DISMISSED** [review]

**Comment by jakelishman** [comment]
@hhorii: we can't drop support for a Python version in a patch release - this should wait for Aer 0.11.

**Pull request merged**